frontend/coreproject/src/lib/components/pages/anime_info/AnimeMainDetails.svelte

```diff
@@ -1,20 +1,29 @@
 <script lang="ts">
 	export let anime: any;
 
-	import { truncate_str } from '$lib/helpers/truncate_str';
 	// icons
 	import PlayCircle from '$icons/PlayCircle.svelte';
-	import Book from '$icons/Book.svelte';
-	import HeadPhones from '$icons/HeadPhones.svelte';
+	import Read from '$icons/Read.svelte';
+	import Listen from '$icons/Listen.svelte';
 	import Video from '$icons/Video.svelte';
-	import Edit3 from '$icons/Edit3.svelte';
 	import Download from '$icons/Download.svelte';
 	import Share from '$icons/Share.svelte';
-  import SettingsOutline from '$icons/SettingsOutline.svelte';
-  import TrendingUp from '$icons/Trending-Up.svelte';
+	import SettingsOutline from '$icons/SettingsOutline.svelte';
+	import TrendingUp from '$icons/Trending-Up.svelte';
 	import Star from '$icons/Star.svelte';
-  import Edit from '$icons/Edit.svelte';
-  import ExternalLink from '$icons/ExternalLink.svelte';
+	import Edit from '$icons/Edit.svelte';
+	import ExternalLink from '$icons/ExternalLink.svelte';
+
+	const roundedRating = Math.floor(anime.rating);
+	const grayStars = 5 - roundedRating;
+
+	const ratingPercentage = Math.ceil((anime.rating / 5) * 100);
+
+	const k_formatter = (total_rating: number) => {
+		return Math.abs(total_rating) > 999
+			? Math.sign(total_rating) * Number((Math.abs(total_rating) / 1000).toFixed(1)) + 'k'
+			: Math.sign(total_rating) * Math.abs(total_rating);
+	};
```

Comment:
What is this?

Reply:
To format a number greater than thousand into something like --  `2.5k`

Reply:
> To format a number greater than thousand into something like -- 2.5k

It's called naturalization.. You shouldn't handle this yourself

Reply:
Oky then should I remove this for now